### PR TITLE
Experiment: Add a simple price filter block with directives

### PR DESCRIPTION
**Pull request by DAreRodz**
## What?

This is a simple reimplementation of the Price Filter block using [directives](https://github.com/WordPress/block-hydration-experiments/issues/80) to filter a list of products by price.

Tracking issue: https://github.com/woocommerce/woocommerce/issues/42486

## Why?

The idea is to test and showcase how a block to filter a list of products would be implemented with directives. Also, to ease a future transition to this new API in case it finally becomes standard.

## How?

This PR continues the work on https://github.com/woocommerce/woocommerce-blocks/pull/8026 and implements a new block called Simple Price Filter. The most interesting changes are in:
- [/assets/js/blocks/simple-price-filter/view.js](https://github.com/woocommerce/woocommerce-blocks/blob/try/add-simple-price-filter-block/assets/js/blocks/simple-price-filter/view.js) (state and logic)
- [/src/BlockTypes/SimplePriceFilter.php](https://github.com/woocommerce/woocommerce-blocks/blob/try/add-simple-price-filter-block/src/BlockTypes/SimplePriceFilter.php) (server-side rendering)

## Video

This is a video recording that shows the block in action. You can see that functionality and appearance is slightly different than the original block, but I think is close enough for this demo.

The URL changes, and the Products list is refreshed, but the page is not reloaded.

https://user-images.githubusercontent.com/6917969/213175525-cc1fce2d-f4a3-444d-9c66-dda67a8c213a.mov

## Next steps
- [x] Get a similar appearance as the current Price Filter block
- [x] Put an initial range of prices (this can be hardcoded for now).
- [x] Fix server-side rendering for the state
- [ ] Handle currencies properly

### CODE REVIEW

File: bin/webpack-helpers.js [hunk 1]
```diff
@@ -63,6 +63,10 @@ const getAlias = ( options = {} ) => {
 			__dirname,
 			`../assets/js/${ pathPart }base/utils/`
 		),
+		'@woocommerce/base-interactivity': path.resolve(
+			__dirname,
+			`../assets/js/${ pathPart }base/interactivity/`
+		),
```
Comment (luisherranz) [inline bin/webpack-helpers.js]: Is this related to this PR or should be part of https://github.com/woocommerce/woocommerce-blocks/pull/8220?

Comment (DAreRodz) [reply bin/webpack-helpers.js]: Oh, nice catch, @luisherranz! That should definitely go into #8220. I'll do a cherry-pick and move that commit there.

Comment (DAreRodz) [reply bin/webpack-helpers.js]: There was a commit removing that code; I reverted it. 👉 https://github.com/woocommerce/woocommerce-blocks/pull/8220/commits/1d135d4eb270c0da9a98295ecd1133a5d9780f58

Comment (DAreRodz) [reply bin/webpack-helpers.js]: Not sure why the changes still appear here, though, now that the same code is in the base branch. 🤔

Comment (luisherranz) [reply bin/webpack-helpers.js]: No problem :)

**Comment by gziolo** [comment]
This bot is making a lot of noise about todo items 😅

**Comment by DAreRodz** [comment]
> This bot is making a lot of noise about todo items 😅

I know 😓 I'll put extra care not to push TODO comments the next time. In the end, those messages only make sense in the original repository, so removing them here is no problem.

**Comment by DAreRodz** [comment]
Closing this PR as I created a new one for this demo: https://github.com/woocommerce/woocommerce-blocks/pull/9926

**Pull request closed without merging**